### ISSUE TITLE
[Feature Request] Filter collection by tags

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#15881]
The current filter by identifier is ok, but sometimes I'd also like to filter by other properties such as presence or absence of a specific (name)tag, since I've gone to the effort to tag my datasets really nicely :) 

I thought I could do this via Apply Rules, but, the tags aren't available there either (nor database build/format/etc.)

**Comment by hexylena** [comment]
Oh my word I didn't know that button existed, wow. Is there a reason (performance? UX cleanliness?) we don't expose all metadata automatically and have it annotated as the correct columns for the output collection?

**Comment by mvdbeek** [comment]
I think it might be a UX / API stability type of thing: If you have a rule that generates a 3rd column, but now that 3rd column contains tag values the rule would break. That's just a guess though. Considering there are other metadata values I think it make sense to start with a minimal set, but I agree that maybe you wouldn't necessarily look into Column -> Add Column from Metadata ?  Or maybe that menu needs to be subdivided into "rules that generate new column" and "Add columns from data" or something like that ?

**Comment by hexylena** [comment]
Yeah, ok, that makes a lot of sense, lacking something more stable like 'named' columns that could be referenced. Perhaps a minimal set would indicate to users that other values are possible (like tags) but just not there by default? I don't know what's the right answer here. Dividing could make sense, or maybe just adding [dropdown section dividers/headers](https://getbootstrap.com/docs/4.6/components/dropdowns/#headers) with "From Data" and "Generate New"?
